src/stirling/source_connectors/socket_tracer/uprobe_manager.cc

```diff
@@ -467,13 +476,19 @@ StatusOr<int> UProbeManager::AttachOpenSSLUProbesOnStaticBinary(const uint32_t p
   return kOpenSSLUProbes.size();
 }
 
-StatusOr<int> UProbeManager::AttachNodeJsOpenSSLUprobes(const uint32_t pid) {
-  PX_ASSIGN_OR_RETURN(const std::filesystem::path proc_exe, proc_parser_->GetExePath(pid));
-
+StatusOr<int> UProbeManager::AttachNodeJsOpenSSLUprobes(const uint32_t pid,
+                                                        const std::filesystem::path& proc_exe) {
   if (DetectApplication(proc_exe) != Application::kNode) {
     return 0;
   }
 
+  const std::string exe_cmdline = proc_parser_->GetPIDCmdline(pid);
+  const std::string node_application_filepath = GetNodeApplicationFilename(exe_cmdline);
+  if (std::find(uprobe_opt_out_.begin(), uprobe_opt_out_.end(), node_application_filepath) !=
```

Comment:
If you convert uprobe_opt_out_ to a `flat_hash_set` (see comment below), then you can use the `contains()` function to make this more concise.

Reply:
Sounds good. This was done in https://github.com/pixie-io/pixie/pull/1971/commits/1207906d583090c861d06971e20a6da9c38d69fd